lib.php

```diff
@@ -347,10 +355,11 @@ function local_augmented_teacher_send_notloggedin_reminder_message() {
                         $log->timecreated = time();
                         $DB->insert_record('local_augmented_teacher_log', $log);
 
-                        // Set sent status
-                        if ($has_sent === false) {
-                            $has_sent = true;
-                        }
+                        $DB->insert_record('local_augmented_teacher_sent', (object)[
```

Comment:
The record should be inserted with an entity in a repository.

Example:
$repo = $base_factory->etc()->etc()->get_repository();
$repo->insert($entity);